resources/assets/coffee/osu_common.coffee

```diff
@@ -235,6 +235,26 @@
   transChoice: (key, count, replacements) ->
     message = Lang.choice key, count, replacements, currentLocale
 
+  uuid: ->
+    Turbolinks.uuid() # no point rolling our own
+
+  updateQueryString: (key, value, url = window.location.href) ->
+    regExp = new RegExp("([?&])" + key + "=.*?(&|#|$)(.*)", "gi")
+    hash = url.split('#')
+
+    if (regExp.test(url))
+      if (value?)
+        return url.replace(regExp, '$1' + key + "=" + value + '$2$3')
```

Comment:
value needs escaping? And may update anchor part as well because it's `url`?

Also, string interpolation. And some more below.